libraries/ui-library-angular/src/lib/link/six-router-link.directive.ts

```diff
@@ -0,0 +1,30 @@
+import { Attribute, Directive, ElementRef, OnChanges, Renderer2, SimpleChanges } from '@angular/core';
+import { ActivatedRoute, Router, RouterLink } from '@angular/router';
+import { LocationStrategy } from '@angular/common';
+
+@Directive({
+  selector: 'six-sidebar-item[routerLink],six-button[routerLink][type=link]',
+})
+export class SixRouterLinkDirective extends RouterLink implements OnChanges {
```

Comment:
ionic is using a similar approach: https://github.com/ionic-team/ionic-framework/blob/main/packages/angular/common/src/directives/navigation/router-link-delegate.ts.

We might want to try to to use the same approach as it seems to work well for them.